### DISCUSSION TITLE
next/link active styling

### DISCUSSION THREAD

**Discussion by rdylina** [vercel/next.js#45173] [Ideas]
### Describe the feature you'd like to request

It would eliminate a lot of heartburn if the Link component had a parameter for active styling. 

Right now you have to do useQuery and deconstruct the query parameters and then manually compare it to each and every link to determine if you should apply active styling. 

### Describe the solution you'd like

<Link href="/" activeStyle={{backgroundColor:"blue"}} className="text-center mt-2"> Click me</Link>

or equivalent would be the best solution in my opinion. This mimics the functionality already in the react router.

### Describe alternatives you've considered

No alternative comes to mind other than the manual check on every link.

**Comment by guitheengineer** [comment]
What do you mean by "active styling"? If it's related to applying some styles when the link is active you can simply check if the href of the link is equal to the current `asPath`

```jsx
import { useRouter } from 'next/router'

function ActiveLink({ children, href }) {
  const router = useRouter()
  const isActive = router.asPath === href
 
 // do something conditionally with isActive
}
```

**Comment by rdylina** [comment]
Doing all that extra BS on every single link in the Nav is exactly what my request is intended to avoid. Please review the proposed suggestion.

Best,
Robert Dylina
Branch Manager NMLS#967879
Talk/Text: 209-233-0121
________________________________

**Comment by guitheengineer** [comment]
> Doing all that extra BS on every single link in the Nav is exactly what my request is intended to avoid.

It's a component, you can reuse it.

**Comment by xriter** [comment]
The way react-router implemented this is pretty nice. It'd be nice if <Link> had something like this too.

```
// see https://reactrouter.com/en/main/components/nav-link
<BaseNavLink
        ref={ref}
        {...props}
        className={({ isActive }) =>
          [
            props.className,
            isActive ? activeClassName : null,
          ]
            .filter(Boolean)
            .join(" ")
        }
        style={({ isActive }) => ({
          ...props.style,
          ...(isActive ? activeStyle : null),
        })}
      />
```

**Comment by andreasmatuszewski** [comment]
Push!
